test/integration/agent/route_test.go

```diff
@@ -121,23 +125,56 @@ func TestInitialize(t *testing.T) {
 			expectNoTrackRules:   true,
 			expectUDPPortInRules: 4789,
 		},
+		{
+			networkConfig: &config.NetworkConfig{
+				TrafficEncapMode: config.TrafficEncapModeNoEncap,
+			},
+			xtablesHoldDuration: 5 * time.Second,
+			expectNoTrackRules:  false,
+		},
 	}
 
 	for _, tc := range tcs {
 		t.Logf("Running Initialize test with mode %s node config %s", tc.networkConfig.TrafficEncapMode, nodeConfig)
+		if tc.xtablesHoldDuration > 0 {
+			closeFn, err := iptables.Lock(iptables.XtablesLockFilePath, 1*time.Second)
+			require.NoError(t, err)
+			go func() {
+				time.Sleep(tc.xtablesHoldDuration)
+				closeFn()
+			}()
+		}
 		routeClient, err := route.NewClient(serviceCIDR, tc.networkConfig, tc.noSNAT)
 		if err != nil {
 			t.Error(err)
 		}
-		if err := routeClient.Initialize(nodeConfig); err != nil {
+		inited1 := make(chan struct{})
+		if err := routeClient.Initialize(nodeConfig, func() {
+			close(inited1)
+		}); err != nil {
 			t.Error(err)
 		}
 
+		select {
+		case <-time.After(tc.xtablesHoldDuration + 5*time.Second):
```

Comment:
when `xtablesHoldDuration` is > 0, should we add a check to verify that `Initialize` does not return immediately?